src/server/nuclearnet/web_socket_proxy_nuclearnet_server.ts

```diff
@@ -200,22 +203,52 @@ class PacketProcessor {
   }
 
   private maybeSendNextPacket() {
-    if (this.outgoingPackets < this.outgoingLimit) {
-      const next = this.queue.pop()
-      if (next) {
+    const next = this.queue.pop()
+    if (next) {
+      const { event, packet } = next
+      const key = `${event}:${packet.peer.name}:${packet.peer.address}:${packet.peer.port}`
+      if (this.canSendEvent(key)) {
         const { event, packet } = next
         let isDone = false
-        const done = () => {
+        const stats = this.getStatsForEvent(key)
+        stats.active++
+        const start = this.clock.performanceNow()
+        const done = (processingTime?: number) => {
           if (!isDone) {
-            this.outgoingPackets--
+            // Update our performance tracking information
+            stats.active -= 1
+            if (processingTime != null) {
+              // Calculate network latency
+              const latency = this.clock.performanceNow() - start
+              const n = stats.active
+              stats.latency = (stats.latency * n + latency) / (n + 1)
+              stats.processingTime = (stats.processingTime * n + processingTime) / (n + 1)
```

Comment:
This is supposed to be an exponential average where n is some constant which defines the inertia of the moving average. It was previously 10 in the prior PR as a magic number

Reply:
Is there something here I can link to? https://en.wikipedia.org/wiki/Moving_average

Reply:
https://en.wikipedia.org/wiki/Exponential_smoothing would do it